ingestion/src/main/java/feast/store/serving/cassandra/FeatureRowToCassandraMutationDoFn.java

```diff
@@ -0,0 +1,73 @@
+package feast.store.serving.cassandra;
+
+import com.google.protobuf.Duration;
+import com.google.protobuf.util.Timestamps;
+import feast.core.FeatureSetProto.FeatureSetSpec;
+import feast.core.FeatureSetProto.FeatureSpec;
+import feast.types.FeatureRowProto.FeatureRow;
+import feast.types.FieldProto.Field;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.slf4j.Logger;
+
+public class FeatureRowToCassandraMutationDoFn extends DoFn<FeatureRow, CassandraMutation> {
+
+  private static final Logger log = org.slf4j.LoggerFactory
+      .getLogger(FeatureRowToCassandraMutationDoFn.class);
+  private Map<String, FeatureSetSpec> featureSetSpecs;
+  private Map<String, Integer> maxAges;
+
+  public FeatureRowToCassandraMutationDoFn(Map<String, FeatureSetSpec> specs, Duration defaultTtl) {
+    this.featureSetSpecs = specs;
+    this.maxAges = new HashMap<>();
+    for (FeatureSetSpec spec : specs.values()) {
+      String featureSetName = spec.getName() + ":" + spec.getVersion();
+      if (spec.getMaxAge() != null && spec.getMaxAge().getSeconds() > 0) {
+        maxAges.put(featureSetName, Math.toIntExact(spec.getMaxAge().getSeconds()));
```

Comment:
Just one thing: even though its a field in the feature set, our intention for maxAge is to be a query time concept (that can change, depending on the user's use case), rather than an ingestion time one, so I'm not entirely sure if using it as the TTL is the best choice.

Reply:
It makes sense about maxAge being a query time concept, where each team using the same features may have different expiration values. 

How would you propose we do this? Redis currently sets TTL to null (no TTL). Do you have any thoughts or work in progress regarding this?

Reply:
> It makes sense about maxAge being a query time concept, where each team using the same features may have different expiration values.
> 
> How would you propose we do this? Redis currently sets TTL to null (no TTL). Do you have any thoughts or work in progress regarding this?

Seems like two options
1. Don't expire (this will accumulate bad state over time)
2. Expire, but force users to reingest

Is it possible to leave this up to whoever sets up the infra, or does it have to be defined as part of schemas?

Reply:
We configure our redis deployments as LRU caches in lieu of a TTL, but afaik there isn't such an option for Cassandra :/ There isn't really an alternative here I guess. It shouldn't be a problem as long as its documented that the feature set default max age is used for TTL when using this store.

Reply:
I'll think about where to add the documentation for this. 

Not sure if a good idea to not have the TTL setup by infra. Some features such as embeddings may not want expiration for a long time.

Reply:
I've added some documentation in Store.proto. I think it makes sense to put it there as users of the Cassandra would check the proto file to see the required and supported configurations in Cassandra anyway.